### PR TITLE
Remove global_name from User objects to comply with OpenAPI spec UserPIIResponse and UserResponse objects

### PR DESCRIPTION
**Pull request by b-rad15**
The spec as defined by https://github.com/discord/discord-api-spec/blob/main/specs/openapi.json
Resolves issue #313 
Wasn't sure how to mark it as intentionally left out so let me know if I should change that (or if anyone else wants to just fix it go for it). I also have generated patch files that could revert this once discord actually implements global_name fully, though I assume those could also be generate after the fact just from this pr.

### CODE REVIEW

**Comment by MazeXP** [comment]
The PR looks good to me!

Remark: The field is only not provided for webhooks as Discord is treating them not a valid user, not because it is not implemented

Discord docs under the definition of message object:
> The author object follows the structure of the user object, but is only a valid user in the case where the message is generated by a user or bot user. If the message is generated by a webhook, the author object corresponds to the webhook's id, username, and avatar. You can tell if a message is generated by a webhook by checking for the webhook_id on the message object.

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IPartialUser.cs [hunk 1]
```diff
@@ -42,7 +42,7 @@ public interface IPartialUser
     Optional<ushort> Discriminator { get; }
 
     /// <inheritdoc cref="IUser.GlobalName" />
-    Optional<string?> GlobalName { get; }
+    // Optional<string?> GlobalName { get; }
```
Comment (Nihlus) [inline Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IPartialUser.cs:45]: I'm not super keen on leaving code commented out like this - preferably, you'd leave an explanation at the class/interface level documentation instead in a `<remarks>` section.

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IUser.cs [hunk 2]
```diff
@@ -55,7 +55,11 @@ public interface IUser : IPartialUser
     /// <summary>
     /// Gets the user's display name, if it is set. For bots, this is the application name.
     /// </summary>
-    new string? GlobalName { get; }
+    /// <remarks>
+    /// Intentionally not implemented as this field is not present in Discord's OpenAPI specification.
```
Comment (Nihlus) [inline Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IUser.cs:59]: This, for example, is a good text you can move there instead.

File: Backend/Remora.Discord.API/API/Gateway/Events/Users/UserUpdate.cs [hunk 3]
```diff
@@ -35,7 +35,8 @@ public record UserUpdate
     Snowflake ID,
     string Username,
     ushort Discriminator,
-    string? GlobalName,
+    // GlobalName is not present in the OpenAPI spec, only in documentation and is not returned by all endpoints
```
Comment (Nihlus) [inline Backend/Remora.Discord.API/API/Gateway/Events/Users/UserUpdate.cs:38]: No need to comment the individual constructor parameters.

**Review by Nihlus — CHANGES_REQUESTED** [review]

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IUser.cs (hunk 2 above)
Comment (b-rad15) [reply Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IUser.cs]: Put this on the remarks for IUser and used inheritdoc to copy just that remark to the other classes/interfaces I removed global_name from

File: Backend/Remora.Discord.API/API/Gateway/Events/Users/UserUpdate.cs (hunk 3 above)
Comment (b-rad15) [reply Backend/Remora.Discord.API/API/Gateway/Events/Users/UserUpdate.cs]: removed these too

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IPartialUser.cs (hunk 1 above)
Comment (b-rad15) [reply Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IPartialUser.cs]: removed these and set the remarks with inheritdoc to IUser's remarks

**Comment by aPinat** [comment]
I feel like there should've been a different solution than removing it outright, as this name as a user's display name is commonly used and there is no other way to get it now.

Discord themselves [say](https://github.com/discord/discord-api-spec#spec-conventions) to follow the docs instead of the OpenAPI spec as well, as it is still in preview.

> This is a preview and it may be not correct. If you find discrepancies between the spec and our [docs](https://discord.com/developers/docs), other than the ones mentioned below, let us know, and follow the docs, not the spec.

**Comment by aPinat** [comment]
I feel like there should've been a different solution for #313 than removing it outright, as this name as a user's display name is commonly used and there is no other way to get it now.

Discord themselves [say](https://github.com/discord/discord-api-spec#spec-conventions) to follow the docs instead of the OpenAPI spec as well, as it is still in preview.

> This is a preview and it may be not correct. If you find discrepancies between the spec and our [docs](https://discord.com/developers/docs), other than the ones mentioned below, let us know, and follow the docs, not the spec.

**Pull request merged**